unit_tests/OrthoTree/morton_codes.cpp

```diff
@@ -2,36 +2,54 @@
 // unit test of morton_codes.h
 
 #include "gtest/gtest.h"
+
 #include "OrthoTree/MortonHelper.h"
+#include "OrthoTree/Types.h"
+
 #include <algorithm>
 #include <cstdint>
 #include <bitset>
 
+using namespace ippl;
 
-TEST(MortonCodesTest, Encode3D) {
+TEST(MortonCodesTest, Encode3D)
+{
+  static constexpr size_t Dim = 3;
   const size_t max_depth = 8;
-  Morton<3> morton(max_depth);
+  Morton<Dim> morton(max_depth);
 
-  std::array<int, 3> coords = { 1, 2, 4 };
-  int64_t code = morton.encode(coords, 3);
-  int64_t expected = 0b1000100010011;
+  grid_coordinate_template<Dim> coords = { 1, 2, 4 };
+  morton_code code = morton.encode(coords, 3);
+  morton_code expected = 0b1000100010011;
   EXPECT_EQ(code, expected);
 }
 
 TEST(MortonCodesTest, Decode3D) {
+  static constexpr size_t Dim = 3;
   const size_t max_depth = 8;
-  Morton<3> morton(max_depth);
+  Morton<Dim> morton(max_depth);
+
+  morton_code code = 0b1000100010011;
+  Morton<Dim>::grid_coordinate coords = morton.decode(code);
+  Morton<Dim>::grid_coordinate expected = { 1, 2, 4 };
+
+  bool is_same = true;
```

Comment:
I don't see a reason for the elementwise comparison. The EXPECT_EQ should do the same as these are std::array

Reply:
nope.. grid coordinate is ippl::Vector, which does not support the comparison (or i used it wrong). I thought about maybe adding a utils.h to the unit_tests where we can overload/provide some basic functions for testing.

Reply:
i created an [issue](#5 ) for further discussion